libfwupdplugin/fu-coswid-firmware.c

```diff
@@ -226,10 +245,17 @@ fu_coswid_firmware_parse_link(FuCoswidFirmware *self,
 				return FALSE;
 			}
 		} else if (tag_id == FU_COSWID_TAG_REL) {
-			if (cbor_isa_negint(pairs[i].value))
+			if (cbor_isa_negint(pairs[i].value)) {
 				link->rel = (-1) - cbor_get_uint8(pairs[i].value);
-			else
+			} else if (cbor_isa_uint(item)) {
 				link->rel = cbor_get_uint8(pairs[i].value);
+			} else {
+				g_set_error_literal(error,
+						    FWUPD_ERROR,
+						    FWUPD_ERROR_INVALID_DATA,
+						    "failed to parse link rel: item is not a uint");
```

Comment:
this error is wrong, you match either negative int or uint.  How about "item is not an integer"

Reply:
agree, fixed.